sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -182,6 +186,9 @@ class ParquetFileFormat
     val parquetOptions = new ParquetOptions(options, sparkSession.sessionState.conf)
     val datetimeRebaseModeInRead = parquetOptions.datetimeRebaseModeInRead
     val int96RebaseModeInRead = parquetOptions.int96RebaseModeInRead
+    // whether read the complete row groups in advance (filter the row groups later if need)
+    // this can avoid reading the footer twice (currently only optimize for vectorized read)
+    val readRowGroupsEarly = enableVectorizedReader
```

Comment:
now only optimize for vectorized read. 
non-vectorized reader use api `org.apache.hadoop.mapreduce.RecordReader#initialize` , I can't pass the footer parameter.